haystack/components/readers/extractive.py

```diff
@@ -83,12 +83,12 @@ def __init__(
             both of these answers could be kept if this variable is set to 0.24 or lower.
             If None is provided then all answers are kept.
         :param model_kwargs: Additional keyword arguments passed to `AutoModelForQuestionAnswering.from_pretrained`
-            when loading the model specified in `model_name_or_path`. For details on what kwargs you can pass,
+            when loading the model specified in `model`. For details on what kwargs you can pass,
             see the model's documentation.
         """
         torch_and_transformers_import.check()
-        self.model_name_or_path = str(model_name_or_path)
-        self.model = None
+        self.model = str(model)
+        self._model = None
```

Comment:
I'd keep the internal names/private variables the same as they are more explicit.